sql-plugin/src/main/scala/com/nvidia/spark/rapids/AvroDataFileReader.scala

```diff
@@ -135,25 +135,15 @@ class AvroDataFileReader(si: SeekableInput) extends AutoCloseable {
   private val sin = new SeekableInputStream(si)
   sin.seek(0) // seek to the start of file and get some meta info.
   private var vin: BinaryDecoder = DecoderFactory.get.binaryDecoder(sin, vin);
-  private var header: Header = null
   private var firstBlockStart: Long = 0
```

Comment:
This is only here to setup the header size and is a var despite only being set once during initialize.  That means we're effectively storing it twice as `firstBlockStart` and `headerSize`.  I suggest removing `firstBlockStart` and setting both `header` and `headerSize` together like this:
```scala
val (header, headerSize) = initialize()

[...]

private def initialize(): (Header, Long) = {
```

Reply:
Good suggestion